### DISCUSSION TITLE
Issue with Remote Desktop Session Freezing When Internet Connection Changes

### DISCUSSION THREAD

**Discussion by amoljagdalepucsd** [Ylianst/MeshCentral#5617] [Q&A]
Hi Community , 

I've encountered an issue while using the desktop session on a remote system. Initially, the session works smoothly. However, when I switch the internet connection of the remote system, the session screen freezes. The window remains visible, but the functionality stops.

Upon investigation, I noticed that I'm receiving a getnetworkinfo message object from the server. Additionally, in the agentcore.c file on the agent side, it detects an ipaddresschanged event and attempts to reconnect.

I'm wondering if this freezing behavior is the default behavior of the system. I'm currently examining the code to find a solution that would allow the session to continue without freezing when the internet connection changes.

Any guidance or suggestions on how to handle this issue within the code would be greatly appreciated

**Comment by si458** [comment]
this is expected behaviour,
this is because the way it relays is relays from you to server then server to remote
when remote drops/changes, the remote reconnects to server as expected
BUT
you are still relaying to the server and the server is trying to relay to the OLD remote and doesnt realise it dropped

so the best way to fix this is simply disconnect wait about 5 seconds then connect, and its all fixed!

its the same sorta issue we have when a user logs out/logs in in windows, 
the remote agent detects the change but the web ui doesnt realise, so it just looks like its frozen when it fact just simple, disconnect, reconnect and away you go!

**Reply by amoljagdalepucsd** [reply]
Thank you @si458 for the details
